src/App.jsx

```diff
@@ -14,45 +16,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const SORT_BY_ALPHABET = 'alphabet';
+const SORT_BY_LENGTH = 'length';
+const SORT_BY_INDEX_DESC = 'reverse default';
+const ASC = 'asc';
+const DESC = 'desc';
+
+function getSortGoods(goods, sortField, sortDirection) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SORT_BY_ALPHABET:
+          return sortDirection === ASC
+            ? good1.localeCompare(good2)
+            : good2.localeCompare(good1);
+
+        case SORT_BY_LENGTH:
+          // eslint-disable-next-line no-nested-ternary
+          return sortDirection === ASC
+            ? good1.length - good2.length
+            : (good1.length === good2.length)
+              ? good2.localeCompare(good1)
+              : good2.length - good1.length;
+
+        case SORT_BY_INDEX_DESC:
+          return preparedGoods.indexOf(good2) - preparedGoods.indexOf(good1);
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [sortDirection, setSortDirection] = useState(ASC);
+  const sortGoods = getSortGoods(goodsFromServer, sortField, sortDirection);
+  const sortReverse = () => {
+    if (sortField === '') {
+      setSortField(SORT_BY_INDEX_DESC);
+    }
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+    if (sortDirection === DESC && sortField === SORT_BY_INDEX_DESC) {
+      setSortField('');
+    }
+
+    if (sortDirection === DESC) {
+      setSortDirection(ASC);
+    }
+
+    if (sortDirection === ASC) {
+      setSortDirection(DESC);
+    }
+  };
+
+  const forReset = () => {
```

Comment:
The function name 'forReset' is not clear. Consider renaming it to 'resetSort' or another more descriptive name.